pydantic_ai_slim/pydantic_ai/_output.py

```diff
@@ -523,6 +550,18 @@ def __init__(
     def mode(self) -> OutputMode:
         return 'tool'
 
+    def dump(self) -> JsonSchema:
+        if self.toolset is None:
+            # need to check expected behavior
+            raise NotImplementedError()
+        processors: list[ObjectOutputProcessor[OutputDataT]] = []
+        for tool_def in self.toolset._tool_defs:  # pyright: ignore [reportPrivateUsage]
+            processor = copy.copy(self.toolset.processors[tool_def.name])
+            processor.object_def.name = tool_def.name
+            processor.object_def.description = tool_def.description
+            processors.append(processor)
+        return UnionOutputProcessor(processors).object_def.json_schema
```

Comment:
Hmm this is a bit ugly :) Can we add a new `json_schema` property to `OutputToolset`, and then generate this inside `OutputToolset.build`, where we already have the processors etc?